### PR TITLE
Small typo (perserve vs preserve)

### PR DESCRIPTION
**Pull request by wschoot**
## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/config/cc_set_hostname.py [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "distros": [ALL_DISTROS],
     "frequency": frequency,
     "examples": [
-        "perserve_hostname: true",
+        "preserve_hostname: true",
```
Comment (blackboxsw) [inline cloudinit/config/cc_set_hostname.py:56]: D'oh, thank you for this catch. Small typo, but big gap in our unittests. I'm sad out unittests aren't strictly validating the individual keys provided in the examples. I think we need to tweak our [examples unittest for this](https://github.com/canonical/cloud-init/blob/main/tests/unittests/config/test_schema.py#L392-L409).

@wschoot if you get a chance to sign our CLA then I can take your commit as is and merge it. https://ubuntu.com/legal/contributors

Ultimately I'd want to fix our unittests here too.
Here's a diff proposal that just highlighted at least 9 other bugs in 9 other documented examples too by making our full schema in unit tests strict to a given module under test.
```diffdiff --git a/tests/unittests/config/test_schema.py b/tests/unittests/config/test_schema.py
index a367cda1c..c31efcc10 100644
--- a/tests/unittests/config/test_schema.py
+++ b/tests/unittests/config/test_schema.py
@@ -406,6 +406,15 @@ class TestCloudConfigExamples:
         """
         schema = get_schema()
         config_load = safe_load(example)
+        # cloud-init-schema is permissive of additionalProperties at the
+        # top-level.
+        # To validate specific schemas against known documented examples
+        # we need to only define the specific module schema and supply
+        # strict=True.
+        # TODO(Drop to pop/update once full schema is strict)
+        schema.pop("allOf")
+        schema.update(schema["$defs"][schema_id])
+        schema["additionalProperties"] = False
         validate_cloudconfig_schema(config_load, schema, strict=True)
 ```

If you'd be willing to sign the CLA and can provide you with the unittest coverage and schema fixes for those 9 errors in this PR.

Comment (blackboxsw) [reply cloudinit/config/cc_set_hostname.py:56]: To validate failures locally once this patch is applied on a system with tox installed:
```
tox -e py3 tests/unittests/config/test_schema.py
```

**Comment by blackboxsw** [comment]
I've confirmed @wschoot has signed Canonical's CLA.
Please also add the following diff to this PR to unlock CI via our github actions:
```diff
diff --git a/tools/.github-cla-signers b/tools/.github-cla-signers
index 746cec5b6..5f54d1062 100644
--- a/tools/.github-cla-signers
+++ b/tools/.github-cla-signers
@@ -97,6 +97,7 @@ vorlonofportland
 vteratipally
 Vultaire
 WebSpider
+wschoot
 xiachen-rh
 xnox
 yangzz-97
```

**Comment by wschoot** [comment]
Hey @blackboxsw 

Thanks for taking the time to show me the ropes for these kind of things. Actually I was just in a rush, spotted a typo, made a PR and wasn't really into writing test and all. I'm more of an ops guy than dev so I figured I just point other people in the right direction and make it easier with a PR and all, instead of just an "issue" for this. Kinda the same as with my earlier "contribution" on https://github.com/canonical/cloud-init/pull/1168. All very small things, strictly documentation too. I see the point of CLA's but personally for me, it's just too much for these documentation errors :) Hope you still like and accept my contributions though!

**Comment by blackboxsw** [comment]
+1 here @wschoot and thanks, I'll land this PR as is given that context and provide a separate one to handle some inconsistencies in our examples more broadly. I'll CC you on that PR just for giggles if interested, but you highlighted a good short-fall here in the testing we have. thanks

**Review by blackboxsw — APPROVED** [review]
Keep up the finds and fixed @wschoot thanks for the contributions here. I'll put up a separate PR now to resolve inconsistencies in our example docs and highlight strict failures in the doc examples.

**Comment by wschoot** [comment]
Awesome! Perhaps you can inclde https://github.com/canonical/cloud-init/pull/1168 in that PR too as it's also just a simple documentation error that was automatically closed (after being approved, weirdly enough :-))

**Comment by holmanb** [comment]
> Awesome! Perhaps you can inclde #1168 in that PR too as it's also just a simple documentation error that was automatically closed (after being approved, weirdly enough :-))

Looks like #1168 involved a change to a file that was removed in dee8231d.

**Comment by blackboxsw** [comment]
added the following PR for the other doc example misses and strict testing https://github.com/canonical/cloud-init/pull/1397

**Pull request merged**